src/components/screens/editor/editor-screen-drawer.tsx

```diff
@@ -0,0 +1,270 @@
+            onChange={ onInputFieldChange }
+            name="formName"
+          />
+          <TextField
+            fullWidth
+            label={ strings.editorScreen.drawer.formUrl }
+            value={ formSettings.formUrl }
+            onChange={ onInputFieldChange }
+            name="formUrl"
+          />
+        </Stack>
+      </Box>
+    );
+  };
+
+  /**
+   * Renders Drawer template section
+   */
+  const renderDrawerTemplateSection = () => {
+    return (
+      <Box
+        sx={{ padding: 2 }}
+      >
+        <Stack spacing={2}>
+          <FormLabel required>{ strings.editorScreen.drawer.formTemplate }</FormLabel>
+          <RadioGroup
+            value={ formSettings.formTemplate }
+            defaultValue={ true }
+            onChange={ onInputFieldChange }
+            name="formTemplate"
+          >
+            <FormControlLabel value={ false } control={ <Radio/> } label={ strings.editorScreen.drawer.formTemplateCustom }/>
+            <FormHelperText>
+              { strings.editorScreen.drawer.formTemplateCustomHelper }
+            </FormHelperText>
+            <FormControlLabel value={ true } control={ <Radio/> } label={ strings.editorScreen.drawer.formTemplateSosmeta }/>
+            <FormHelperText>
+              { strings.editorScreen.drawer.formTemplateSosmetaHelper }
+            </FormHelperText>
+          </RadioGroup>
+          <TextField
+            fullWidth
+            label={ strings.editorScreen.drawer.formTemplateSchema }
+            value={ formSettings.formSchema }
+            disabled={ !formSettings.formTemplate }
+            onChange={ onInputFieldChange }
+            name="formSchema"
+          />
+        </Stack>
+      </Box>
+    );
+  };
+
+  /**
+   * Renders Drawer identification section
+   */
+  const renderDrawerIdentificationSection = () => {
+    return (
+      <Box
+        sx={{ padding: 2 }}
+      >
+        <Stack spacing={2}>
+          <FormLabel required>{ strings.editorScreen.drawer.formIdentification }</FormLabel>
+          <RadioGroup
+            value={ formSettings.formIdentification }
+            defaultValue={ true }
+            onChange={ onInputFieldChange }
+            name="formIdentification"
+          >
+            <FormControlLabel value={ true } control={ <Radio/> } label={ strings.editorScreen.drawer.formIdentificationService }/>
+            <FormHelperText>
+              { strings.editorScreen.drawer.formIdentificationHelper }
+            </FormHelperText>
+            <FormControlLabel value={ false } control={ <Radio/> } label={ strings.editorScreen.drawer.formIdentificationNone }/>
+          </RadioGroup>
+        </Stack>
+      </Box>
+    );
+  };
+
+  useEffect(() => {
+    const templateSectionValid = formSettings.formTemplate ? formSettings.formTemplate && !!formSettings.formSchema : !formSettings.formTemplate;
+    setValid(!!formSettings.formName && templateSectionValid);
+  }, [formSettings]);
```

Comment:
For readability, consider making methods that you run inside useEffect(), don't write logic inside it.

Reply:
Done